middle_end/flambda/simplify_primitives.ml

```diff
@@ -120,15 +120,15 @@ let primitive (p : Clambda_primitives.primitive) (args, approxs)
     A.value_block tag (Array.of_list approxs), C.Benefit.zero
   | Praise _ ->
     expr, A.value_bottom, C.Benefit.zero
-  | Pmakearray(_, _) when is_empty approxs ->
+  | Pmakearray(_, _, Alloc_heap) when is_empty approxs ->
     Prim (Pmakeblock(0, Asttypes.Immutable, Some [], Alloc_heap), [], dbg),
     A.value_block (Tag.create_exn 0) [||], C.Benefit.zero
-  | Pmakearray (Pfloatarray, Mutable) ->
+  | Pmakearray (Pfloatarray, Mutable, Alloc_heap) ->
```

Comment:
It's not obvious to me that this is safe. If a local value is given an approximation that uniquely identifies it, is there a risk that it could be CSE'd with some heap value that has the same contents and thus have its lifetime extended beyond its region?

Reply:
Thanks, that makes sense. I'll do these simplifications regardless of mode, but we'll need to be careful with Flambda 2.